libraries/ts/src/margin/marginAccount.ts

```diff
@@ -120,11 +120,38 @@ export class MarginAccount {
   get isBeingLiquidated() {
     return !this.info?.marginAccount.liquidation.equals(PublicKey.default)
   }
-  /** A number where 1 and above is subject to liquidation and 0 is no leverage. */
+  /** A qualitative measure of the the health of a margin account.
+   * A higher value means more risk in a qualitative sense.
+   * Properties:
+   *  non-negative, range is [0, infinity)
+   *  zero only when an account has no exposure at all
+   *  account is subject to liquidation at a value of one
+  */
   get riskIndicator() {
-    const requiredCollateral = this.valuation.requiredCollateral.asNumber()
-    const effectiveCollateral = this.valuation.effectiveCollateral.asNumber()
-    return effectiveCollateral === 0 ? 0 : requiredCollateral / effectiveCollateral
+    return this.computeRiskIndicator(
+      this.valuation.requiredCollateral.asNumber(),
+      this.valuation.weightedCollateral.asNumber(),
+      this.valuation.liabilities.asNumber(),
+    )
+  }
+
+  /** A just-okay risk indicator (TODO improve me) */
+  computeRiskIndicator(
+    requiredCollateral: number,
+    weightedCollateral: number,
+    liabilities: number,
+  ): number {
+    if (requiredCollateral < 0) throw "requiredCollateral must be non-negative"
+    if (weightedCollateral < 0) throw "weightedCollateral must be non-negative"
+    if (liabilities < 0) throw "liabilities must be non-negative"
```

Comment:
It's good hygiene to throw something deriving Error
```suggestion
    if (requiredCollateral < 0) throw new Error("requiredCollateral must be non-negative")
    if (weightedCollateral < 0) throw new Error("weightedCollateral must be non-negative")
    if (liabilities < 0) throw new Error("liabilities must be non-negative")
```